File: docs/src/cpp/sort_with_lambda.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
  vector<string> cities{"arta", "ioannina", "preveza", "igoumenitsa"};
  sort(cities.begin(), cities.end(),
       [](auto a, auto b) { return a.length() < b.length(); });
  for (auto city : cities) {
    cout << city << endl;
  }
}

File: docs/src/java/sort_with_java/SortWithLambda.java
package sort_with_java;

import java.util.*;

public class SortWithLambda {
    public static void main(String[] args) {
        List<String> cities = new ArrayList<>(List.of("arta", "ioannina", "preveza", "igoumenitsa"));
        cities.sort((a, b) -> a.length() - b.length());
        for (String city : cities) {
            System.out.println(city);
        }
    }
}
